File: java2csharp/java/Samsung/Look.EdgeScreen/feeds/CocktailFeedsProvider.java

package com.example.edgescreen.feeds;

import com.example.edgescreen.R;
import com.samsung.android.sdk.look.cocktailbar.SlookCocktailManager;
import com.samsung.android.sdk.look.cocktailbar.SlookCocktailProvider;

import android.content.Context;
import android.widget.RemoteViews;

import java.text.SimpleDateFormat;
import java.util.Date;

public class CocktailFeedsProvider extends SlookCocktailProvider {
    @Override
    public void onUpdate(Context context, SlookCocktailManager cocktailBarManager, int[] cocktailIds) {
        // create RemoteViews
        RemoteViews rv = new RemoteViews(context.getPackageName(), R.layout.feeds_panel_layout);
        String str = context.getResources().getString(R.string.feeds_contents);

        // set text in RemoteViews
        rv.setTextViewText(R.id.text, str + " at " + getCurrentDate());

        // update cocktail
        for (int i = 0; i < cocktailIds.length; i++) {
            cocktailBarManager.updateCocktail(cocktailIds[i], rv);
        }
    }

    private String getCurrentDate() {
        long now = System.currentTimeMillis();
        Date date = new Date(now);

        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
        return format.format(date);
    }
}

File: java2csharp/csharp/Samsung/Look.EdgeScreen/feeds/CocktailFeedsProvider.cs
using System;

namespace com.example.edgescreen.feeds
{

	using SlookCocktailManager = com.samsung.android.sdk.look.cocktailbar.SlookCocktailManager;
	using SlookCocktailProvider = com.samsung.android.sdk.look.cocktailbar.SlookCocktailProvider;

	using Context = android.content.Context;
	using RemoteViews = android.widget.RemoteViews;


	public class CocktailFeedsProvider : SlookCocktailProvider
	{
		public override void onUpdate(Context context, SlookCocktailManager cocktailBarManager, int[] cocktailIds)
		{
			// create RemoteViews
			RemoteViews rv = new RemoteViews(context.PackageName, R.layout.feeds_panel_layout);
			string str = context.Resources.getString(R.@string.feeds_contents);

			// set text in RemoteViews
			rv.setTextViewText(R.id.text, str + " at " + CurrentDate);

			// update cocktail
			for (int i = 0; i < cocktailIds.Length; i++)
			{
				cocktailBarManager.updateCocktail(cocktailIds[i], rv);
			}
		}

		private string CurrentDate
		{
			get
			{
				long now = DateTimeHelperClass.CurrentUnixTimeMillis();
				DateTime date = new DateTime(now);
    
				SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
				return format.format(date);
			}
		}
	}

}